doc/developer-notes.md

```diff
@@ -839,6 +840,37 @@ namespace {
 #endif // BITCOIN_FOO_BAR_H
 ```
 
+Construct On First Use
+------------------------
+
+It is not possible to control the order in which non-local static objects in *different* translation units are initialized.
+This is the so-called "static initialization order problem". It could cause the use of an object:
+- before it is constructed (on program startup)
+- after it is destructed (at program exit)
+
+Both cases lead to undefined behavior.
+
+To prevent the static initialization order problem, use the Construct On First Use:
+
+```C++
+BCLog::Logger& LogInstance()
```

Comment:
is this code snippet a good explanation of the concept?

Reply:
Why not? I'd say "example" rather "explanation".

Reply:
Sure; people can look it up anyway for more info.